File: Majority Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int,int> mp;
        
        for(int i=0;i<nums.size();i++)
            mp[nums[i]]++;
        
        int n = nums.size();
       
        for(int i=0;i<nums.size();i++){
            if(mp[nums[i]] > n/2)
                return nums[i];
       
        }
        
        return 0;
    }
};

//Another way is to use Moore's Voting Algorithm

File: Majority Element.java
class Solution 
{
    public int majorityElement(int[] nums) 
    {
        HashMap<Integer, Integer> map = new HashMap<>();
        
        int len = nums.length;
        
        for(int n: nums)
        {
            int count =  map.getOrDefault(n,0);
            count++;
            
            map.put(n,count);
            
            //As soon as the count of number increases by half of numbers in array; we return the number.
            if(count>(len/2)) return n;
        }
        return -1;
    }
}